src/sentry/uptime/endpoints/organization_uptime_stats.py

```diff
@@ -0,0 +1,170 @@
+import logging
+from collections import defaultdict
+
+from drf_spectacular.utils import extend_schema
+from google.protobuf.timestamp_pb2 import Timestamp
+from rest_framework.request import Request
+from rest_framework.response import Response
+from sentry_protos.snuba.v1.endpoint_time_series_pb2 import TimeSeriesRequest, TimeSeriesResponse
+from sentry_protos.snuba.v1.request_common_pb2 import RequestMeta, TraceItemType
+from sentry_protos.snuba.v1.trace_item_attribute_pb2 import (
+    AttributeAggregation,
+    AttributeKey,
+    AttributeValue,
+    Function,
+    StrArray,
+)
+from sentry_protos.snuba.v1.trace_item_filter_pb2 import ComparisonFilter, TraceItemFilter
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import StatsArgsDict, StatsMixin, region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.uptime.models import ProjectUptimeSubscription
+from sentry.utils.snuba_rpc import timeseries_rpc
+
+logger = logging.getLogger(__name__)
+
+
+MAX_UPTIME_SUBSCRIPTION_IDS = 100
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Uptime Monitors"])
+class OrganizationUptimeStatsEndpoint(OrganizationEndpoint, StatsMixin):
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.CRONS
+    permission_classes = (OrganizationPermission,)
+
+    def get(self, request: Request, organization: Organization) -> Response:
```

Comment:
It might be good to type the response so that this auto documents

Reply:
i've left the API experimental for now. can follow up w/ documentation, i need to implement some serializers for that.